xdsl/dialects/riscv.py

```diff
@@ -1001,6 +1237,28 @@ def verify_(self) -> None:
     def assembly_line_args(self) -> tuple[_AssemblyInstructionArg, ...]:
         return self.rd, self.csr, self.immediate
 
+    def print_attributes(self, printer: Printer) -> None:
+        printer.print(" ")
+        printer.print(self.csr.value.data)
+        if self.writeonly is not None:
+            printer.print(" w")
+        if self.immediate is not None:
+            printer.print(" ")
+            printer.print(self.immediate.value.data)
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> dict[str, Attribute]:
+        attributes = dict[str, Attribute]()
+        if (csr := parser.parse_optional_integer()) is not None:
+            attributes["csr"] = IntegerAttr(csr, i32)
+        else:
+            raise NotImplementedError()
+        if (_ := parser.parse_optional_characters("w")) is not None:
```

Comment:
You don't need the walrus operator here at all if you discard the result, simply do `if parser.parse_optional_characters("w") is not None:`

Reply:
Done